### ISSUE TITLE
`MaybeValid<T>`

### ISSUE THREAD

**Issue by joshlf** [google/zerocopy#1833]
A type with the same layout as `T`, but no bit validity requirements.

**Comment by kupiakos** [comment]
Yes, essentially it acts as a deferred `TryFromBytes`. It also makes any external type `FromBytes` like `MaybeUninit` can, but conversion to the inner type remains unsafe.

- `MaybeValid<T>: FromBytes`
- `MaybeValid<T>: IntoBytes where T: IntoBytes`
- `MaybeValid<T>: DerefMut<Target = T> where T: FromBytes`
- `T: TryFrom<MaybeValid<T>> where T: TryFromBytes`
- `&T: TryFrom<MaybeValid<&T>> where T: TryFromBytes`
- `&mut T: TryFrom<MaybeValid<&mut T>> where T: TryFromBytes`
- `T: From<MaybeValid<T>> where T: FromBytes`
- `&T: From<MaybeValid<&T>> where T: FromBytes`
- `&mut T: From<MaybeValid<&mut T>> where T: FromBytes`

Most of these conversions are `unsafe` on `MaybeUninit`.
